### DISCUSSION TITLE
A way to output a msg to a specific partition

### DISCUSSION THREAD

**Discussion by rangeldaykes** [rabbitmq/rabbitmq-stream-dotnet-client#268] [Q&A]
is there a way to output a msg to a specific partition, or just with the HashRoutingMurmurStrategy?

**Comment by Gsantomaggio** [comment]
I am not sure to understand the question.
Can you elaborate more?
Do you want to publish a message to a specific partition? 
The `HashRoutingMurmurStrategy` is used in the super stream.

**Reply by rangeldaykes** [reply]
Thanks for the answer
yes i would like to control which partition the msg goes to
for example to divide my partitions by geographic region, or by Client, guaranteeing the order of the messages in the partition
I thought I could configure how the producer would do the routing

**Comment by Gsantomaggio** [comment]
You can define the routing client side. See the [example](https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/tree/main/docs/SuperStream)  



Particularly here: https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/blob/main/docs/SuperStream/SuperStreamProducer.cs#L43

```csharp
                   SuperStreamConfig = new SuperStreamConfig() // <2>
                    {
                        Routing = msg => msg.Properties.MessageId.ToString() //<-- here you can define your routing 
                    }
```

**Reply by rangeldaykes** [reply]
ok, but will a hash algorithm be applied on top of this messageID?

**Reply by Gsantomaggio** [reply]
Yes, here https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/blob/main/RabbitMQ.Stream.Client/RawSuperStreamProducer.cs#L321 

but you can change the function. For Example, in your case, it could be like:

```csharp
                   SuperStreamConfig = new SuperStreamConfig() 
                    {
                       Routing = msg => msg.ApplicationProperties["region"].ToString() 
                    }
```

**Comment by Gsantomaggio** [comment]
@rangeldaykes ok I got what you mean. And currently, the client does not implement Key strategy. We will add it.

**Comment by Gsantomaggio** [comment]
@rangeldaykes FYI https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/issues/269

**Reply by rangeldaykes** [reply]
Thanks @Gsantomaggio

**Comment by Gsantomaggio** [comment]
https://www.nuget.org/packages/RabbitMQ.Stream.Client/1.4.1 contains the feature
